cookbook/session/avoid_session_start.rst

```diff
@@ -0,0 +1,54 @@
+.. index::
+    single: Sessions, cookies
+
+Avoid Starting Sessions for Anonymous Users
+===========================================
+
+Sessions in Symfony applications are automatically started whenever they are necessary.
+This includes writing in the user's session, creating a flash message and logging
+in users. In order to start the session, Symfony creates a cookie which will be
+added to every user request.
+
+However, there are other scenarios when a session is started automatically and a
+cookie will be created even for anonymous users. First, consider the following
+template code commonly used to display flash messages:
+
+.. code-block:: html+jinja
+
+    {% for flashMessage in app.session.flashbag.get('notice') %}
+        <div class="flash-notice">
+            {{ flashMessage }}
+        </div>
+    {% endfor %}
+
+Even if the user is not logged in and even if you haven't created any flash message,
+just calling the ``get()`` method of the ``flashbag`` will start a session. This
+may hurt your application performance because all users will receive a session
+cookie. To avoid this behavior, add a check before trying to access the flash messages:
+
+.. code-block:: html+jinja
+
+    {% if app.session.started %}
+        {% for flashMessage in app.session.flashbag.get('notice') %}
+            <div class="flash-notice">
+                {{ flashMessage }}
+            </div>
+        {% endfor %}
+    {% endif %}
+
+Another scenario where session cookies will be automatically sent is when the
+requested URL is covered by a firewall, even when anonymous users can access
+to that URL:
+
+.. code-block:: yaml
+
+    # app/config/security.yml
+    security:
+        firewalls:
+            main:
+                pattern:    ^/
+                form_login: ~
+                anonymous:  ~
+
+This behavior is caused because in Symfony applications, anonymous users are
+technically authenticated.
```

Comment:
This is a killer, but I'm not sure it's true (yay!). I just tested locally on a 2.6 project, and once I protected against the flash messages (using `app.request.hasPreviousSession`) and removed some session checks from my user-land code, there was no session cookie.

The security-related session stuff is handled in ContextListener. On `kernel.request`, it correctly doesn't start the session unless there was a previous session (https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Security/Http/Firewall/ContextListener.php#L76). Then, on `kernel.response`, it correctly doesn't save the token to the session if we're dealing with an AnonymousToken: https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Security/Http/Firewall/ContextListener.php#L125

So for me, this note is not valid - but I wonder where you got this idea from @javiereguiluz? Is there something else?